app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java

```diff
@@ -0,0 +1,264 @@
+                LOGGER.info("Preference changed which is not a ListPreference, and not the display name");
+                // For all other preferences, set the summary to the value's
+                // simple string representation.
+                preference.setSummary(stringValue);
+            }
+            return true;
+        }
+
+        /**
+         * If the key is display_name this method is called to update the summary
+         * and the Profile in the database.
+         *
+         * @param preference The preference which is changed
+         * @param stringValue The string value which is new
+         */
+        private void updateDisplayName(final Preference preference, final String stringValue) {
+            if (firstLoad) {
+                firstLoad = false;
+
+                try {
+                    preference.setSummary(serviceLocator.getDatabase().getProfiles().get(0).getName());
+                    return;
+                } catch (IOException e) {
+                    LOGGER.error("Couldn't get profiles from database while loading for the first time", e);
+                }
+
+                preference.setSummary(stringValue);
+            } else {
+                try {
+                    LOGGER.info("Updating profile name");
+                    final Profile profile = serviceLocator.getDatabase().getProfiles().get(0);
+                    serviceLocator.getDatabase().updateProfile(profile, new Profile(stringValue, profile.getKeyPair()));
+                } catch (IOException e) {
+                    LOGGER.error("Couldn't get profiles from database", e);
+                }
+
+                preference.setSummary(stringValue);
+            }
+        }
+    };
+
+    /**
+     * Binds a preference's summary to its value. More specifically, when the
+     * preference's value is changed, its summary (line of text below the
+     * preference title) is updated to reflect the value. The summary is also
+     * immediately updated upon calling this method. The exact display format is
+     * dependent on the type of preference.
+     *
+     * @see #sBindPreferenceSummaryToValueListener
+     */
+    private static void bindPreferenceSummaryToValue(final Preference preference) {
+        // Set the listener to watch for value changes.
+        preference.setOnPreferenceChangeListener(sBindPreferenceSummaryToValueListener);
+
+        // Trigger the listener immediately with the preference's
+        // current value.
+        sBindPreferenceSummaryToValueListener.onPreferenceChange(preference,
+                PreferenceManager
+                        .getDefaultSharedPreferences(preference.getContext())
+                        .getString(preference.getKey(), ""));
+    }
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setupActionBar();
+
+        final Intent intent = getIntent();
+        if (serviceLocator == null) {
```

Comment:
Because this method is also run when opening a fragment, and it should only be set the first time the SettingsActivity is created.